src/rust/engine/logging/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+// Enable all clippy lints except for many of the pedantic ones. It's a shame this needs to be copied and pasted across crates, but there doesn't appear to be a way to include inner attributes from a common source.
+#![cfg_attr(
+  feature = "cargo-clippy",
+  deny(
+    clippy,
+    default_trait_access,
+    expl_impl_clone_on_copy,
+    if_not_else,
+    needless_continue,
+    single_match_else,
+    unseparated_literal_suffix,
+    used_underscore_binding
+  )
+)]
+// It is often more clear to show that nothing is being moved.
+#![cfg_attr(feature = "cargo-clippy", allow(match_ref_pats))]
+// Subjective style.
+#![cfg_attr(
+  feature = "cargo-clippy",
+  allow(len_without_is_empty, redundant_field_names)
+)]
+// Default isn't as big a deal as people seem to think it is.
+#![cfg_attr(
+  feature = "cargo-clippy",
+  allow(new_without_default, new_without_default_derive)
+)]
+// Arc<Mutex> can be more clear than needing to grok Orderings:
+#![cfg_attr(feature = "cargo-clippy", allow(mutex_atomic))]
+// We only use unsafe pointer derefrences in our no_mangle exposed API, but it is nicer to list
+// just the one minor call as unsafe, than to mark the whole function as unsafe which may hide
+// other unsafeness.
+#![cfg_attr(feature = "cargo-clippy", allow(not_unsafe_ptr_arg_deref))]
+
+pub mod logger;
+
+extern crate lazy_static;
+extern crate log;
+extern crate num_enum;
+extern crate parking_lot;
+extern crate simplelog;
+extern crate ui;
+
+pub type Logger = logger::Logger;
+
+// TODO My intuition is that these belog here, because putting them outside of the `logging`
```

Comment:
No, this was in response to a comment by @illicitonion with the question of whether this should go on the FFI definition because it's tied to Rust/Python interaction, or whether it should go in the logging lib because it's exclusively relevant to logging. I decided for the latter, and that comment is my reasoning. Can delete it, though.

Reply:
Relevant comment: https://github.com/pantsbuild/pants/pull/6817#discussion_r242117796